Tests/SentryTests/Profiling/SentryProfilerSwiftTests.swift

```diff
@@ -0,0 +1,373 @@
+
+    private class Fixture {
+        lazy var options: Options = {
+            let options = Options()
+            options.dsn = SentryProfilerSwiftTests.dsnAsString
+            return options
+        }()
+        lazy var client: TestClient! = TestClient(options: options)
+        lazy var hub: SentryHub = {
+            let hub = SentryHub(client: client, andScope: scope)
+            hub.bindClient(client)
+            Dynamic(hub).tracesSampler.random = TestRandom(value: 1.0)
+            Dynamic(hub).profilesSampler.random = TestRandom(value: 0.5)
+            return hub
+        }()
+        let scope = Scope()
+        let message = "some message"
+        let transactionName = "Some Transaction"
+        let transactionOperation = "Some Operation"
+    }
+
+    private var fixture: Fixture!
+
+    override func setUp() {
+        super.setUp()
+        fixture = Fixture()
+        SentryTracer.resetAppStartMeasurementRead()
+    }
+
+    override func tearDown() {
+        super.tearDown()
+        clearTestState()
+        SentryTracer.resetAppStartMeasurementRead()
+#if os(iOS) || os(tvOS) || targetEnvironment(macCatalyst)
+        SentryFramesTracker.sharedInstance().resetFrames()
+        SentryFramesTracker.sharedInstance().stop()
+#endif
+    }
+
+    // This test is only available on newer APIs because of the availability of `DispatchQueue.activate`
+    @available(tvOS 10.0, *)
+    @available(OSX 10.12, *)
+    @available(iOS 10.0, *)
+    func testConcurrentProfilingTransactions() {
+        let options = fixture.options
+        options.profilesSampleRate = 1.0
+        options.tracesSampleRate = 1.0
+
+        let queue = DispatchQueue(label: "SentryProfilerSwiftTests", attributes: [.concurrent, .initiallyInactive])
+        let group = DispatchGroup()
+
+        let numberOfTransactions = 10
+        for _ in 0 ..< numberOfTransactions {
+            group.enter()
+            let exp = expectation(description: "finished span")
+            let span = fixture.hub.startTransaction(name: fixture.transactionName, operation: fixture.transactionOperation)
+
+            // Some busy work to try and get it to show up in the profile.
+            let str = "a"
+            var concatStr = ""
+            for _ in 0..<100_000 {
+                concatStr = concatStr.appending(str)
+            }
+
+            queue.asyncAfter(deadline: .now() + 2) {
+                span.finish()
+                exp.fulfill()
+                group.leave()
+            }
+        }
+
+        queue.activate()
+        group.wait()
+        waitForExpectations(timeout: 5)
+
+        guard let envelope = self.fixture.client.captureEnvelopeInvocations.first else {
+            XCTFail("Expected to capture at least 1 event")
+            return
+        }
+        XCTAssertEqual(1, envelope.items.count)
+        guard let profileItem = envelope.items.first else {
+            XCTFail("Expected at least 1 additional envelope item")
```

Comment:
`l`: We check exactly for one envelope item above.
```suggestion
            XCTFail("Expected 1 additional envelope item")
```

Reply:
Nice catch. I'll do this separately, since these are no longer additional items, but first-class items in their own envelope now.